tablet_test.go

```diff
@@ -0,0 +1,207 @@
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		-6917529027641081857,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 9}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		-4611686018427387905,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 8}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		-2305843009213693953,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 9}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		-1,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 8}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		2305843009213693951,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 3}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		4611686018427387903,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 3}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		6917529027641081855,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 7}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test1",
+		TimeUUID(),
+		9223372036854775807,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 7}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		-6917529027641081857,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 9}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		-4611686018427387905,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 8}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		-2305843009213693953,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 9}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		-1,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 8}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		2305843009213693951,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 3}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		4611686018427387903,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 3}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		6917529027641081855,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 7}},
+		"",
+	},
+	{
+		sync.RWMutex{},
+		"test2",
+		TimeUUID(),
+		9223372036854775807,
+		"table1",
+		8,
+		nil,
+		[]ReplicaInfo{{TimeUUID(), 7}},
+		"",
+	},
+}
+
+func TestFindTablets(t *testing.T) {
+	id := findTablets(tablets, "test1", "table1")
+	assertEqual(t, "id", 0, id)
+
+	id = findTablets(tablets, "test2", "table1")
+	assertEqual(t, "id", 8, id)
+
+	id = findTablets(tablets, "test3", "table1")
+	assertEqual(t, "id", -1, id)
+}
+
+func TestFindTabletForToken(t *testing.T) {
+	tablet := findTabletForToken(tablets, parseInt64Token("0"), 0)
+	assertTrue(t, "tablet.lastToken >= 0", tablet.lastToken >= 0)
+
+	tablet = findTabletForToken(tablets, parseInt64Token("9223372036854775807"), 0)
+	assertTrue(t, "tablet.lastToken >= 9223372036854775807", tablet.lastToken >= 9223372036854775807)
```

Comment:
You could probably make this assertion more specific: `tablet.lastToken == 9223372036854775807` since the tablet list is hardcoded and we know exactly the tablet that should be returned. I'd also check `value - 1`, `value + 1` to make sure there are no off-by-one errors in binary search.

Reply:
Added one test with the lower limit of the range.